### PR TITLE
Improve error messages for invalid config files

### LINKED ISSUE #1020: Improve error messages for invalid config files
It's too easy to make one too many closing braces or forget one in a YANG instance data file (conf file).  We should make the error messages better.  Currently they look like this:

```
Stack Traceback
===============
(1) Lua function ‘?’ at file ‘core/main.lua:168’ (best guess)
    Local variables:
     reason = string: “core/main.lua:26: unrecognized parameter: queue”
     (*temporary) = C function: print
(2) global C function ‘error’
(3) Lua global ‘assert’ at file ‘core/main.lua:26’
    Local variables:
     v = nil
(4) Lua field ‘parse’ at file ‘lib/yang/data.lua:389’
    Local variables:
     P = table: 0x41c524c8  {line:22, line_pos:445, column:9, str:softwire-config {
 instance {
   device 0000:05:00.0;
   queue 
...
```

As you can see, we do have the precise position of the error; in that case (and especially for initial configurations) we should not produce a backtrace and instead produce the error directly.  Also we can give some more context, like "unexpected queue statement at foo:22:9 while parsing queue structure; expecting one of id, foo, bar, or baz."

### PR DESCRIPTION
**Pull request by dpino**
Fixes #1020

### CODE REVIEW

**Comment by dpino** [comment]
Example:

```
$ sudo ./snabb lwaftr run --name lwaftr --cpu 10 --conf lwaftr-64k.conf --on-a-stick 82:00.0
lwaftr-64k.conf: loading compiled configuration from lwaftr-64k.o
lwaftr-64k.conf: compiled configuration is out of date; recompiling.
lwaftr-64k.conf: loading source configuration
      queue1 {
            ^
lwaftr-64k.conf:64049:12: error: unrecognized parameter: queue1
```

**Comment by wingo** [comment]
The file name is available as `stream.io.filename`, so no need to plumb around.  Probably we should add a method on stream to return it.  WDYT?

**Comment by dpino** [comment]
OK, I amended the patches to add a `stream:filename()'  method and use it from Parser, instead of passing `filename` around.

File: src/lib/stream.lua [hunk 1]
```diff
@@ -434,6 +434,8 @@ function Stream:write(...)
    for _, arg in ipairs({ ... }) do write1(self, arg) end
 end
 
+function Stream:filename() return self.io.filename end
```
Comment (wingo) [inline src/lib/stream.lua]: Please add a comment above the function indicating that the result may be nil.

**Review by wingo — APPROVED** [review]
LGTM with one nit

**Pull request merged**